lib/logstash/codecs/avro.rb

```diff
@@ -85,7 +85,7 @@ class LogStash::Codecs::Avro < LogStash::Codecs::Base
   config :target, :validate => :field_reference
 
   def open_and_read(uri_string)
-    open(uri_string).read
+    URI.open(uri_string, &:read)
```

Comment:
review note: in the doc we don't mention about `ftp` URIs. I do think this will be separate from current bug fix change but if we need to support `ftp`, we need to include `net/ftp` dependency as [`open-uri` requires](https://github.com/ruby/open-uri/blob/b41d2c63de32390c127056590e7a3eda263e2eb8/lib/open-uri.rb#L755-L757).